packages/core/src/constructor.ts

```diff
@@ -151,15 +152,21 @@ export function instantiateInteractor<E extends Element, F extends Filters<E>, A
     },
 
     perform<T>(fn: (element: E) => T): Interaction<T> {
-      return interaction(`run perform on ${description(options)}`, () => converge(() => fn(resolver(options))))
+      return interaction(
+        `run perform on ${description(options)}`,
+        () => converge(() => fn(resolver(options))),
+        serializeActionOptions({ type: "interaction", actionName: "perform", options })
+      )
     },
 
     assert(fn: (element: E) => void): ReadonlyInteraction<void> {
-      return check(`${description(options)} asserts`, () => {
-        return converge(() => {
-          fn(resolver(options));
-        });
-      });
+      return check(
+        interaction(
+          `${description(options)} asserts`,
+          () => converge(() => { fn(resolver(options)) }),
+          serializeActionOptions({ type: "check", actionName: "assert", options }),
```

Comment:
Can we just add these options directly to the `interaction` method and have it call `serializeActionOptions` inside itself?

Reply:
Done